lib/modules.nix

```
@@ -451,6 +476,7 @@ rec {
           imports = m.imports or [];
           options = m.options or {};
           config = addFreeformType (addMeta (m.config or {}));
+          class = m.class or null;
```

```suggestion
          class = m._class or null;
```

This feels like something that end users shouldn't set, so maybe it should be indicated as internal.

---

I've changed it to a non-shorthand-only attribute, just like `meta`. I don't think it needs to be internal, even if it's possibly rare for users to interact with it. Marking this with an underscore would lead to users taking the underscore less seriously.

---

> just like `meta`

`meta` (and the deprecated `require`) are the only examples of attributes only available in non-shorthand syntax. And `meta` is only that way because [originally](https://github.com/NixOS/nixpkgs/pull/15025) it was `config.meta`.

At least one problem with supporting attributes only in non-shorthand syntax is that it has a poor error messages:

```nix
error: The option `class' does not exist
```

Because of this I'd still prefer `_class`, but then make it work in both syntaxes

---

Done.